pkg/controller/atlasproject/alert_configurations.go

```diff
@@ -317,3 +337,35 @@ func isAlertConfigSpecEqualToAtlas(logger *zap.SugaredLogger, alertConfigSpec md
 
 	return true
 }
+
+func canAlertConfigurationReconcile(ctx context.Context, atlasClient mongodbatlas.Client, protected bool, project *mdbv1.AtlasProject, logger *zap.SugaredLogger) (bool, error) {
+	if !protected {
+		return true, nil
+	}
+
+	latestConfig := &mdbv1.AtlasProjectSpec{}
+	latestConfigString, ok := project.Annotations[customresource.AnnotationLastAppliedConfiguration]
+	if ok {
+		if err := json.Unmarshal([]byte(latestConfigString), latestConfig); err != nil {
+			return false, err
+		}
+	}
```

Comment:
So for Alter Configs to be managed, not only a config must already exist, but also we should not find certain differences?
Can you elaborate a bit more on this?

Reply:
Most of the resources can't be identified by a single key, many are composed an/or different compositions when they are different types of resources being manipulated.
Some of them had a comparison function built already and in this case, I decided to reuse them to match the same reconciliation logic, for other, I have to come up with the comparison logic